realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1092,6 +1092,20 @@ public boolean isValid() {
         return this;
     }
 
+    /**
+     * Condition that finds a distinct set of objects of a specific class. As a Realm is unordered,
+     * it is undefined which objects are returned in case of multiple occurrences.
+     *
+     * @param fieldName the field name.
+     * @return the query object.
+     * @throws IllegalArgumentException if a field name does not exist or the field is not indexed.
+     */
+    public RealmQuery<E> distinct(String fieldName) {
+        long columnIndices[] = schema.getColumnIndices(fieldName, RealmFieldType.STRING);
```

Comment:
This would only allow String fields, other types are allowed as well:  Boolean, Long, Date